python/modules/IcePy/Types.cpp

```diff
@@ -2108,6 +2108,38 @@ IcePy::SequenceInfo::marshalPrimitiveSequence(const PrimitiveInfoPtr& pi, PyObje
     }
 }
 
+PyObject*
+IcePy::SequenceInfo::createSequenceFromBuffer(const SequenceMappingPtr& sm, const char* buffer, Py_ssize_t size)
+{
+    PyObjectHandle result;
+#if PY_VERSION_HEX >= 0x03030000
+    PyObjectHandle memoryview = PyMemoryView_FromMemory(const_cast<char*>(buffer), size, PyBUF_READ);
+#else
+    PyObjectHandle bytearray = PyByteArray_FromStringAndSize(buffer, size);
```

Comment:
We could instead use PyBuffer_FillInfo & PyMemoryView_FromBuffer to create a Py_buffer from the memory block and the memory view from that buffer.